pixeltable/functions/gemini.py

```diff
@@ -242,6 +247,122 @@ def _(model: str) -> str:
     return f'request-rate:veo:{model}'
 
 
+# The default dimensionality for Gemini embeddings that works with embedding indexes.
+# Google recommends using 768, 1536, or 3072 output dimensions, but 3072 exceeds the max vector length for embedding
+# indexes (2000). It's possible to exceed 2000 as long as the output vector is not stored in an embedding index.
+_DEFAULT_EMBEDDING_DIMENSIONALITY = 1536
+
+
+@pxt.udf(resource_pool='request-rate:gemini', batch_size=32)
+async def generate_embedding(
+    input: Batch[str],
+    *,
+    model: str = 'gemini-embedding-001',
```

Comment:
I'd argue we shouldn't be specifying a default here (the models are continually changing and it's best not to hardcode it - in most UDFs model is a required parameter; we should follow that convention unless there's a specific reason to deviate from it here)